connect/runtime/src/test/java/org/apache/kafka/connect/util/clusters/EmbeddedConnectCluster.java

```diff
@@ -291,6 +293,35 @@ public String configureConnector(String connName, Map<String, String> connConfig
                 "Could not execute PUT request. Error response: " + responseToString(response));
     }
 
+    /**
+     * Validate a given connector configuration. If the configuration validates or
+     * has a configuration error, an instance of {@link Response} is returned. If the validation fails
+     * an exception is thrown.
+     *
+     * @param connClassName the name of the connector class
+     * @param connConfig    the intended configuration
+     * @throws JsonProcessingException if the response failed to deserialize into a {@link ConfigInfos} instance
+     * @throws ConnectRestException if the REST api returns error status
+     * @throws ConnectException if the configuration fails to be serialized or if the request could not be sent
+     */
+    public ConfigInfos validateConnectorConfig(String connClassName, Map<String, String> connConfig)
```

Comment:
minus two lines this method is the same as the above `configureConnector`. Should we refactor the common parts into a `putConnectorConfig` method or something?

Reply:
@kkonstantine Yes we could do that, should we make `putConnectorConfig` a public or private method? If we were to implement the new shared method, I'm thinking it could take a `url` and a `config`, and return a type of `Result`.

Reply:
I'd recommend `protected`, in case we need to subclass in the future. 

Also, I'd say, let's return a `String` from `responseToString(response)` which your new method will parse to get the `ConfigInfos` object then. Otherwise, it's close to what `requestPut` does already. wdyt?

Re: the input arguments, yes I agree, `url` and `config` make sense.

Reply:
I like that approach, updated.